adoc/chapters/programming_interface.adoc

```diff
@@ -6057,57 +6067,6 @@ buffer construction.
 ----
 
 
-==== Shared SYCL ownership of the host memory
-
-When an instance of [code]#std::shared_ptr# is passed to the buffer
-constructor, then the buffer object and the developer's application share
-the memory region. If the shared pointer is still used on the application's
-side then the data will be copied back from the buffer or image and will be
-available to the application after the buffer or image is destroyed.
-
-If the [code]#shared_ptr# is not empty, the contents of the referenced
-memory are used to initialize the buffer.  If the [code]#shared_ptr# is
-empty, then the buffer is created with uninitialized memory.
-
-When the buffer is destroyed and the data have potentially been updated, if
-the number of copies of the shared pointer outside the runtime is 0, there
-is no user-side shared pointer to read the data. Therefore the data is not
-copied out, and the buffer destructor does not need to wait for the data
-processes to be finished, as the outcome is not needed on the application's
-side.
-
-This behavior can be overridden using the [code]#set_final_data()#
-member function of the buffer class, which will by any means force the buffer
-destructor to wait until the data is copied to wherever the
-[code]#set_final_data()# member function has put the data (or not wait nor copy
-if set final data is [code]#nullptr)#.
-
-[source,,linenums]
-----
-{
-  std::shared_ptr<int> ptr { data };
-  {
-    buffer<int, 1> b { ptr, range<2>{ 10, 10 } };
-    // update the data
-    [...]
-  } // Data is copied back because there is an user side shared_ptr
-}
-----
-
-[source,,linenums]
-----
-{
-  std::shared_ptr<int> ptr { data };
-  {
-    buffer<int, 1> b { ptr, range<2>{ 10, 10 } };
-    // update the data
-    [...]
-    ptr.reset();
-  } // Data is not copied back, there is no user side shared_ptr.
-}
-----
-
-
```

Comment:
Should we keep these examples?

Reply:
I have checked again : these were the only 2 code examples showing `sycl::buffer` and `std::shared_ptr`. Why removing them? Are they wrong now?

Reply:
I agree that the examples are good to keep.  I was hoping to move all of the normative description of the buffer APIs to [section 4.7.2 Buffers](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers), and in particular the normative description of the blocking behavior and copy-back behavior to [section 4.7.2.3 Buffer synchronization rules](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:buf-sync-rules).

How do you feel about the following ... I'll still augment section 4.7.2.3 to contain a complete description of the blocking behavior and copy-back behavior, so all of that information is together in one place.  We can still keep section 4.7.4, though, as a set of examples.  In this case, I'd probably retitle the section to something like "Buffer examples", and the introduction of that section will read something like:

> This section provides some examples showing typical use cases for buffer. These examples are intended to clarify the definition of the buffer interfaces, but the content of this section is nonnormative.

Over time, we can add more examples to this section to clarify other behavior about buffers.

Reply:
I restored the examples, and converted section 4.7.4 to an examples section as I describe in the comment above.  See commit 15e8eab47d8198b1bb263da991bf5a7e80990505.

Note that this PR is incomplete at this point because I need answers to the questions in (https://gitlab.khronos.org/sycl/Specification/-/issues/478)